src/libraries/System.Text.Encoding/tests/Ascii/EqualsTests.cs

```diff
@@ -0,0 +1,126 @@
+
+        [Fact]
+        public static void NonEqualValues_AndNonAsciiCharacters_ReturnsFalse() => Assert_NotEqual(127, 128);
+
+        private static void Assert_NotEqual(byte left, byte right)
+        {
+            // Equals
+            // (byte, byte)
+            Assert.False(Ascii.Equals(new byte[] { left }, new byte[] { right })); // non-vectorized code path
+            Assert.False(Ascii.Equals(Enumerable.Repeat(left, 100).ToArray(), Enumerable.Repeat(right, 100).ToArray())); // vectorized code path
+            // (byte, char)
+            Assert.False(Ascii.Equals(new byte[] { left }, new char[] { (char)right }));
+            Assert.False(Ascii.Equals(Enumerable.Repeat(left, 100).ToArray(), Enumerable.Repeat((char)right, 100).ToArray()));
+            // (char, char)
+            Assert.False(Ascii.Equals(new char[] { (char)left }, new char[] { (char)right }));
+            Assert.False(Ascii.Equals(Enumerable.Repeat((char)left, 100).ToArray(), Enumerable.Repeat((char)right, 100).ToArray()));
+
+            // EqualsIgnoreCase
+            // (byte, byte)
+            Assert.False(Ascii.EqualsIgnoreCase(new byte[] { left }, new byte[] { right }));
+            Assert.False(Ascii.EqualsIgnoreCase(Enumerable.Repeat(left, 100).ToArray(), Enumerable.Repeat(right, 100).ToArray()));
+            // (byte, char)
+            Assert.False(Ascii.EqualsIgnoreCase(new byte[] { left }, new char[] { (char)right }));
+            Assert.False(Ascii.EqualsIgnoreCase(Enumerable.Repeat(left, 100).ToArray(), Enumerable.Repeat((char)right, 100).ToArray()));
+            // (char, char)
+            Assert.False(Ascii.EqualsIgnoreCase(new char[] { (char)left }, new char[] { (char)right }));
+            Assert.False(Ascii.EqualsIgnoreCase(Enumerable.Repeat((char)left, 100).ToArray(), Enumerable.Repeat((char)right, 100).ToArray()));
+        }
+
+        public static IEnumerable<object[]> ExactlyTheSame_TestData
+        {
+            get
+            {
+                yield return new object[] { "test", "test" };
+
+                for (char textLength = (char)0; textLength <= 127; textLength++)
+                {
+                    yield return new object[] { new string(textLength, textLength), new string(textLength, textLength) };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ExactlyTheSame_TestData))]
+        public static void ExactlyTheSame_ReturnsTrue(string left, string right)
+        {
+            Assert.True(Ascii.Equals(Encoding.ASCII.GetBytes(left), Encoding.ASCII.GetBytes(right)));
+            Assert.True(Ascii.Equals(Encoding.ASCII.GetBytes(left), right));
+            Assert.True(Ascii.Equals(left, right));
+
+            Assert.True(Ascii.EqualsIgnoreCase(Encoding.ASCII.GetBytes(left), Encoding.ASCII.GetBytes(right)));
+            Assert.True(Ascii.EqualsIgnoreCase(left, right));
+            Assert.True(Ascii.EqualsIgnoreCase(Encoding.ASCII.GetBytes(left), right));
```

Comment:
I'm generally not a fan of tests that validate many different things--in this case, the behavior of 6 different methods/overloads. With a single test here, only the first failure will get reported instead of the test results showing all of the failing asserts.

I recommend having 6 different `Theory` methods here instead--one for each method/overload.